celery/backends/base.py

```diff
@@ -647,10 +648,25 @@ def _forget(self, task_id):
 
     def _store_result(self, task_id, result, state,
                       traceback=None, request=None, **kwargs):
+
+        if state in self.READY_STATES:
+            date_done = datetime.utcnow().isoformat()
+        else:
+            date_done = None
+
         meta = {
-            'status': state, 'result': result, 'traceback': traceback,
+            'status': state,
+            'result': result,
+            'traceback': traceback,
             'children': self.current_task_children(request),
             'task_id': bytes_to_str(task_id),
+            'name': request.task,
+            'args': request.args,
+            'kwargs': request.kwargs,
+            'worker': request.hostname,
+            'date_done': date_done,
```

Comment:
This 'date_done' is kind of hack. Is there a way to access the 'received', 'started', 'finished' values from the task cache from within this backend function?

Reply:
@auvipy is there any way to get this data? It would be useful to have.